cmd/metal-api/internal/datastore/machine.go

```diff
@@ -442,26 +443,49 @@ func (rs *RethinkStore) FindWaitingMachine(partitionid, sizeid string) (*metal.M
 		"state": map[string]string{
 			"value": string(metal.AvailableState),
 		},
-		"waiting": true,
+		"waiting":    true,
+		"liveliness": metal.MachineLivelinessAlive,
 	}).Sample(1)
 
-	var available metal.Machines
-	err := rs.searchEntities(&q, &available)
+	var availables metal.Machines
+	err := rs.searchEntities(&q, &availables)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(available) < 1 {
+	if len(availables) < 1 {
 		return nil, fmt.Errorf("no machine available")
 	}
 
-	// we actually return the machine from the machine table, not from the wait table
-	// otherwise we will get in trouble with update operations on the machine table because
-	// we have mixed timestamps with the entity from the wait table...
-	m, err := rs.FindMachineByID(available[0].ID)
-	if err != nil {
-		return nil, err
-	}
+	return &availables[0], nil
+}
 
-	return m, nil
+func (rs *RethinkStore) EvaluateMachineLiveliness(m metal.Machine) (metal.MachineLiveliness, error) {
```

Comment:
Probably more like a personal thing, dunno. In the past, I tried to limit the scope of the datastore package to only contain pretty dumb CRUD operations. As this function has a little more business logic, I personally find it better suited in the service layer where it was before. Can we move it back now that we do not use it twice anywhere or do you think it makes more sense to move it here?